src/ast2ram/provenance/SubproofGenerator.cpp

```diff
@@ -177,15 +174,30 @@ Own<ram::Operation> SubproofGenerator::generateReturnInstantiatedValues(const as
     VecOwn<ram::Expression> values;
 
     // get all values in the body
-    for (ast::Literal* lit : clause.getBodyLiterals()) {
-        if (auto atom = dynamic_cast<ast::Atom*>(lit)) {
-            for (ast::Argument* arg : atom->getArguments()) {
+    for (const auto* lit : clause.getBodyLiterals()) {
+        if (auto atom = dynamic_cast<const ast::Atom*>(lit)) {
```

Comment:
Please use `as<>`.  One of my PRs went though the whole codebase and replaced `dynamic_cast`.  It'll also make my future work easier ;), and it's shorter to type.